Sources/MapboxNavigation/ArrivalController.swift

```diff
@@ -0,0 +1,172 @@
+
+import UIKit
+import MapboxDirections
+import MapboxCoreNavigation
+import MapboxMobileEvents
+
+/// A component to encapsulate `EndOfRouteViewController` presenting logic such as enabling/disabling, handling autolayout, keyboard, positioning camera, etc.
+class ArrivalController: NavigationComponentDelegate {
+    
+    // MARK: - Properties
+    
+    weak private(set) var navigationViewData: NavigationViewData!
```

Comment:
I’m still concerned about the use of this backpointer to NavigationViewController etc. Can we add methods to NavigationComponentDelegate that return whatever ArrivalController et al. need from those classes?

Reply:
I tried to minimize dependencies through the protocol as much as I could. Currently it boiled down to using `NavigationView` and `Router`. It would be hard (or even impossible?) to get further than that since `controllers` logic is meant to interact with `NavigationView` and ease the burden for `NavigationViewController` by that.

Reply:
I’m pretty sure we’ll need to rethink this approach when we make ArrivalController or any of the other new classes public for use in custom UIs apart from NavigationViewController, since NavigationView is private too. It gets back to the original reason for wanting to get rid of RouteMapViewController, that it encapsulated too much behavior that was only usable as part of NavigationViewController.